### PR TITLE
Prepare release v1.34.0

### PR DESCRIPTION
**Pull request by albertteoh**
Prepares Jaeger UI release v1.34.0

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,20 +1,17 @@
 # Cutting a Jaeger UI release
 
-1. Determine the version for the release.
```
Comment (albertteoh) [inline RELEASE.md:3]: I felt the instructions could be improved to be a bit more succinct.

File: RELEASE.md [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
-1. Create and merge, per approval, a PR which preps the release.
+1. Create and merge, per approval, a PR which preps the release ([example](https://github.com/jaegertracing/jaeger-ui/pull/1767)).
    1. The PR title should match the format "Prepare release vX.Y.Z".
-   1. CHANGELOG.md
-      - Change the version of the current release from "Next (unreleased)" to "vX.Y.Z (Month D, YYYY)" where "vX.Y.Z" is the semver for this release.
-      - Make sure all relevant changes made since the last release are present and listed under the current release.
-        - Tip: Run `node scripts/get-changelog.js` script that prints a dozen most recent commits in the chagelog format.
+   2. CHANGELOG.md
+      - Change the version of the current release from "Next (unreleased)" to "vX.Y.Z (Month D, YYYY)" where "vX.Y.Z" is the [semver](https://semver.org) for this release.
+      - Run `make changelog` to list all changes since the last release.
```
Comment (albertteoh) [inline RELEASE.md:7]: Given the jaeger maintainers are (currently) releasing both jaeger-ui and jaeger, I thought to make the experience more consistent between the two repos, hence the introduction of a Makefile and the same python scripts used to generate the changelog. Please let me know if you disagree and I can revert.

If we agree to this change, I could also remove the existing `get-changelog.js` and `get-tracking-version.js` scripts  _if_ they're not used for any other purpose other than for releases.

File: scripts/release-notes.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                        help='The file containing your personal github token to access the github API. ' +
+                             '(default: ~/.github_token)')
+    parser.add_argument('--repo', type=str, default='jaeger',
+                        help='The repository name to fetch commit logs from. (default: jaeger)')
+    parser.add_argument('--exclude-dependabot', action='store_true',
+                        help='Excludes dependabot commits. (default: false)')
+    parser.add_argument('--num-commits', type=int,
+                        help='Print this number of commits from git log. ' +
+                             '(default: number of commits before the previous tag)')
+    parser.add_argument('--point-markdown-char', type=str, default='*',
```
Comment (albertteoh) [inline scripts/release-notes.py:117]: This script is essentially a copy-paste of https://github.com/jaegertracing/jaeger/blob/main/scripts/release-notes.py with some minor enhancements such as making the point markdown char configurable for consistency with the existing convention, as well as auto-capitalizing sentences in each point, which have been used to generate the above change log.

If we think these improvements are worthwhile keeping, I can sync it up with the `jaeger` repo.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,34 @@
 
 ### Fixes
 
+## v1.34.0 (2023-10-04)
+
+### Enhancements
+
+- Handle the case in which child span starts before and ends after parent span. ([@GLVSKiriti](https://github.com/GLVSKiriti) in [#1780](https://github.com/jaegertracing/jaeger-ui/pull/1780))
```
Comment (yurishkuro) [inline CHANGELOG.md:13]:
```suggestion
- Handle Critical Path computation when child span starts before and ends after parent span. ([@GLVSKiriti](https://github.com/GLVSKiriti) in [#1780](https://github.com/jaegertracing/jaeger-ui/pull/1780))
```

**Review by yurishkuro — COMMENTED** [review]
I just merged a bug fix #1805, please include it in the release

File: scripts/release-notes.py (hunk 3 above)
Comment (yurishkuro) [reply scripts/release-notes.py:117]: I would actually prefer to have just a single copy of the full script in the main repo, and in this repo have a thin wrapper that will download the other script from github.

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,34 @@
...
+- Improve tooltip for tag search ([@yurishkuro](https://github.com/yurishkuro) in [#1799](https://github.com/jaegertracing/jaeger-ui/pull/1799))
+- Fix copyicon to receive reactnode instead of string ([@anshgoyalevil](https://github.com/anshgoyalevil) in [#1791](https://github.com/jaegertracing/jaeger-ui/pull/1791))
+- Refactor critical path code to prevent it from ever throwing exceptions ([@GLVSKiriti](https://github.com/GLVSKiriti) in [#1785](https://github.com/jaegertracing/jaeger-ui/pull/1785))
+- Print suggestion on fixing prettier-lint errors ([@yurishkuro](https://github.com/yurishkuro) in [#1783](https://github.com/jaegertracing/jaeger-ui/pull/1783))
+- Upgrade to node 18 ([@yurishkuro](https://github.com/yurishkuro) in [#1776](https://github.com/jaegertracing/jaeger-ui/pull/1776))
+- Extend try/catch for critical path ([@yurishkuro](https://github.com/yurishkuro) in [#1778](https://github.com/jaegertracing/jaeger-ui/pull/1778))
+- Convert utils/treenode.js and selectors/trace.js to typescript ([@yurishkuro](https://github.com/yurishkuro) in [#1777](https://github.com/jaegertracing/jaeger-ui/pull/1777))
```
Comment (yurishkuro) [inline CHANGELOG.md:35]: Looking at a mishmash of topics here I kinda want to resurrect our attempt at labeling the PRs into categories https://github.com/jaegertracing/jaeger/issues/4799

File: scripts/release-notes.py (hunk 3 above)
Comment (albertteoh) [reply scripts/release-notes.py:117]: Do you think we should do the same for `scripts/draft-release.py` as well? We'd need to make some minor differences  that would need to be either make configurable or opinionated.

File: RELEASE.md (hunk 1 above)
Comment (yurishkuro) [reply RELEASE.md:3]: I don't know, don't think original text hurts, especially if a new person is doing it for the first time

Comment (yurishkuro) [reply RELEASE.md:3]: but I see you copied it below, so good

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.PHONY: changelog
+changelog:
+	./scripts/release-notes.py --exclude-dependabot --repo jaeger-ui --point-markdown-char '-'
```
Comment (yurishkuro) [inline Makefile:3]: could we just curl/wget this from the main repo ?

https://raw.githubusercontent.com/jaegertracing/jaeger/main/scripts/release-notes.py

File: .codecov.yml [hunk 7]
```diff
@@ -32,3 +32,4 @@ coverage:
         paths: null
 ignore:
   - "packages/jaeger-ui/src/utils/helpers/ReactShallowRenderer.js"
+  - "scripts"
```
Comment (yurishkuro) [inline .codecov.yml:35]: I think you want .prettierignore, not codecov

File: Makefile (hunk 6 above)
Comment (albertteoh) [reply Makefile:3]: Yup, that's what I plan to do, just wanted to get your thoughts on whether we should do the same for `./scripts/draft-release.py` as well before I make that change.

Comment (yurishkuro) [reply Makefile:3]: Sure

**Comment by albertteoh** [comment]
I seem to have buggered up the rebase, so I've created a [new PR](https://github.com/jaegertracing/jaeger-ui/pull/1849).

**Pull request closed without merging**